### ISSUE TITLE
Panic occurs in payload validation with OneOf (HTTP)

### ISSUE THREAD

**Issue by ikawaha** [goadesign/goa#3221]
# Bug?

In HTTP transport, the validation of Payload containing `OneOf` is generated as follows:

```Go
// ValidatePetOwnerRequestBody runs the validations defined on
// PetOwnerRequestBody
func ValidatePetOwnerRequestBody(body *PetOwnerRequestBody) (err error) {
	if body.Pet.Type == nil {
		err = goa.MergeErrors(err, goa.MissingFieldError("Type", "body.pet"))
	}
	if body.Pet.Value == nil {
		err = goa.MergeErrors(err, goa.MissingFieldError("Value", "body.pet"))
	}
	if body.Pet.Type != nil {
		if !(*body.Pet.Type == "dog" || *body.Pet.Type == "cat") {
			err = goa.MergeErrors(err, goa.InvalidEnumValueError("body.pet.Type", *body.Pet.Type, []interface{}{"dog", "cat"}))
		}
	}
	return
}
```

The generated verification does not perform a nil check on `body.Pet`, which may cause a panic if the payload is invalid.

# Expected generated codes

```Go
// ValidatePetOwnerRequestBody runs the validations defined on
// PetOwnerRequestBody
func ValidatePetOwnerRequestBody(body *PetOwnerRequestBody) (err error) {
	if body.Pet == nil || body.Pet.Type == nil {
		err = goa.MergeErrors(err, goa.MissingFieldError("Type", "body.pet"))
	}
	if body.Pet == nil || body.Pet.Value == nil {
		err = goa.MergeErrors(err, goa.MissingFieldError("Value", "body.pet"))
	}
	if body.Pet != nil && body.Pet.Type != nil {
		if !(*body.Pet.Type == "dog" || *body.Pet.Type == "cat") {
			err = goa.MergeErrors(err, goa.InvalidEnumValueError("body.pet.Type", *body.Pet.Type, []interface{}{"dog", "cat"}))
		}
	}
	return
}
```

# Sample Minimal Design

```go
package design

import (
	. "goa.design/goa/v3/dsl"
)

var Dog = Type("Dog", func() {
	Description("Dogs are cool")
	Field(1, "Name")
	Field(2, "Breed")
	Required("Name", "Breed")
})

var Cat = Type("Cat", func() {
	Description("Cats are cool too")
	Field(1, "Name")
	Field(2, "Age", Int)
	Required("Name", "Age")
})

var PetOwner = Type("PetOwner", func() {
	Description("A pet owner")
	OneOf("pet", func() {
		Description("The owner's pet")
		Field(1, "dog", Dog)
		Field(2, "cat", Cat)
	})
})

var _ = Service("pet-hotel", func() {
	Method("check-in", func() {
		Payload(func() {
			Field("1", "body", PetOwner)
			Required("body")
		})
		Result(PetOwner)
		HTTP(func() {
			POST("/check-in")
			Response(StatusOK)
		})
		GRPC(func() {
			Response(CodeOK)
		})
	})
})
```

### OK
```shell
curl -X POST localhost:8888/check-in -d '{"body":{"pet":{"Type":"dog","Value":"{\"Name\":\"POCHI\",\"Breed\":\"AKITA\"}"}}}'
{"pet":{"Type":"dog","Value":"{\"Name\":\"POCHI\",\"Breed\":\"AKITA\"}"}}
```

### Missing `body` (Error Response)
```shell
$ curl -X POST localhost:8888/check-in -d '{"pet":{"Type":"dog","Value":"{\"Name\":\"POCHI\",\"Breed\":\"AKITA\"}"}}'
{"name":"missing_field","id":"yLEPEv6N","message":"\"body\" is missing from body","temporary":false,"timeout":false,"fault":false}
```

### Missing `pet` (❗PANIC❗)
```shell
$ curl -X POST localhost:8888/check-in -d '{"body":{"hello":{"Type":"dog","Value":"{\"Name\":\"POCHI\",\"Breed\":\"AKITA\"}"}}}'
curl: (52) Empty reply from server
```

Server log
```shell
[pethotelapi] 16:18:22 id=wHhoJHII req=POST /check-in from=127.0.0.1
2022/12/27 16:18:22 http: panic serving 127.0.0.1:51180: runtime error: invalid memory address or nil pointer dereference
goroutine 13 [running]:
net/http.(*conn).serve.func1()
	/Users/ikawaha/sdk/go1.19.3/src/net/http/server.go:1850 +0xbf
panic({0x152e160, 0x19e6f80})
	/Users/ikawaha/sdk/go1.19.3/src/runtime/panic.go:890 +0x262
goa.design/examples/union/gen/http/pet_hotel/server.ValidatePetOwnerRequestBody(0xc000014170)
	/Users/ikawaha/go/src/github.com/goadesign/examples/union/gen/http/pet_hotel/server/types.go:103 +0x23
goa.design/examples/union/gen/http/pet_hotel/server.ValidateCheckInRequestBody(0xc0000023c0?)
	/Users/ikawaha/go/src/github.com/goadesign/examples/union/gen/http/pet_hotel/server/types.go:93 +0x74
goa.design/examples/union/gen/http/pet_hotel/server.DecodeCheckInRequest.func1(0x16a0740?)
	/Users/ikawaha/go/src/github.com/goadesign/examples/union/gen/http/pet_hotel/server/encode_decode.go:48 +0x136

...snip
```

# Comments

For `Type` and `Value`, it was possible to add a nil check by making requiredValTmpl as follows:

```Go
	requiredValTmpl = `if {{ if .isPointer }}{{ .target }} == nil || {{ end }}{{ $.target }}.{{ .attCtx.Scope.Field $.reqAtt .req true }} == nil {
        err = goa.MergeErrors(err, goa.MissingFieldError("{{ .req }}", {{ printf "%q" $.context }}))
}`
```

However, I could not figure out how to modify the generated code for the enum check. I could not disassemble it because the `target` passed to the template is `body.Pet.Type`.

**Comment by raphael** [comment]
Thank you for the nice bug report, this is fixed in #3222

**Comment by ikawaha** [comment]
Thanks for the quick fix!
